mdata/aggmetric.go

```diff
@@ -92,14 +92,11 @@ func NewAggMetric(store Store, cachePusher cache.CachePusher, key schema.AMKey,
 // Sync the saved state of a chunk by its T0.
 func (a *AggMetric) SyncChunkSaveState(ts uint32, sendPersist bool) ChunkSaveCallback {
 	return func() {
-		a.Lock()
-		if ts > a.lastSaveFinish {
-			a.lastSaveFinish = ts
+		lastSaveStart := atomic.LoadUint32(&a.lastSaveStart)
+		if ts > lastSaveStart {
+			atomic.StoreUint32(&a.lastSaveStart, ts)
```

Comment:
race condition!
I think you need something like
```
        newVal := ts
	prev := atomic.SwapUint32(&a.lastSaveStart, newVal)
	for prev > newVal {
		newVal = prev
		prev = atomic.SwapUint32(&a.lastSaveStart, newVal)
	}
```
(i took this from idx/memory.bumpLastUpdate())

Reply:
good catch, i'll update that